move/sources/canvas/token.move

```diff
@@ -436,6 +445,20 @@ module addr::canvas_token {
         };
     }
 
+    fun assert_canvas_enabled_for_non_admin(
+        caller_addr: address,
+        canvas: Object<Canvas>,
+    ) acquires Canvas {
+        let caller_is_admin = is_admin(canvas, caller_addr);
```

Comment:
there is a constrain that `let caller_is_admin = is_admin(canvas, caller_addr);` must be called before `let canvas_ = borrow_global<Canvas>(object::object_address(&canvas));`, because `is_admin` also borrow that object, although it's borrowing a immutable version, but the compiler still doesn't allow that in different order